### PR TITLE
Add option to ignore default flags. BREAKING

### PR DESCRIPTION
**Pull request by samccone**
Originally from
https://github.com/GoogleChrome/chrome-launcher/pull/38

fixes https://github.com/GoogleChrome/chrome-launcher/issues/28

---
high level

⚠️  Breaking API change
* drop enableExtensions boolean

👁  CL specific changes
* add an option to ignore all non-required flags
* refactor tests to use a common spawn helper

As always thank you for reviewing.

### CODE REVIEW

**Comment by paulirish** [comment]
over in #65 there's more discussion on the shape of the API.  

Right  now there's more interest in passing in a Map of flags ([API here](https://github.com/GoogleChrome/chrome-launcher/issues/65#issuecomment-349161118)) rather than using `ignoreDefaultFlags` in the JS API.

Additionally, we'll need a public `defaultFlags()` method to make the CLI proposal work.

**Comment by paulirish** [comment]
I updated this PR. rebased to master and then added `launcher.defaultFlags()`

**Comment by paulirish** [comment]
Puppeteer has shipped with the ignoreFlags boolean API. They weren't fans of the array/map approach of the [_JS vGamma API_](https://github.com/GoogleChrome/chrome-launcher/issues/65#issuecomment-349161118).

At this point, it probably makes sense to align with them and keep the API boring. @patrickhulce wdyt?

**Comment by patrickhulce** [comment]
> At this point, it probably makes sense to align with them and keep the API boring.

Sigh, 😞  yeah I agree let's just stay aligned

**Comment by paulirish** [comment]
@samccone my changes lgty?

**Comment by samccone** [comment]
yes lgtm.

**Comment by paulirish** [comment]
k. we'll merge. heads up: this will require a 2.0.0 release.

**Comment by samccone** [comment]
@paulirish thoughts?

File: README.md [hunk 1]
```diff
@@ -62,9 +62,9 @@ npm install chrome-launcher
   // Default: 'info'
   logLevel: string;
 
-  // (optional) Enable extension loading
+  // (optional) Flags specific in [flags.ts](https://github.com/GoogleChrome/chrome-launcher/blob/master/flags.ts) will not be included.
   // Default: false
-  enableExtensions: boolean;
+  ignoreDefaultFlags: booelan;
```
Comment (paambaati) [inline README.md]: @samccone Typo in `booelan`.

**Pull request merged**